extension/solarwindsextension/internal/config.go

```diff
@@ -0,0 +1,156 @@
+// Copyright 2024 SolarWinds Worldwide, LLC. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"errors"
+	"fmt"
+	"maps"
+	"slices"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configopaque"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/exporter/otlpexporter"
+)
+
+// Config represents a Solarwinds Extension configuration.
+type Config struct {
+	// DataCenter ID (e.g. na-01).
+	DataCenter string `mapstructure:"data_center"`
+	// IngestionToken is your secret generated SWO ingestion token.
+	IngestionToken configopaque.String `mapstructure:"token"`
+	// CollectorName is the name you will see in the SWO UI
+	CollectorName string `mapstructure:"collector_name"`
+	// Insecure disables TLS in the exporters.
+
+	// ⚠️ Warning: For testing purpose only.
+	// EndpointURLOverride sets OTLP endpoint directly, it overrides the DataCenter configuration.
+	EndpointURLOverride string `mapstructure:"endpoint_url_override"`
+	// ⚠️ Warning: For testing purpose only.
+	// Insecure disables the TLS security. It can be used only together with EndpointURLOverride.
+	Insecure bool `mapstructure:"insecure"`
+}
+
+var (
+	ErrMissingDataCenter    = errors.New("invalid configuration: 'data_center' must be set")
+	ErrMissingToken         = errors.New("invalid configuration: 'token' must be set")
+	ErrMissingCollectorName = errors.New("invalid configuration: 'collector_name' must be set")
+	ErrInsecureInProd       = errors.New("invalid configuration: 'insecure' is not allowed in production mode")
+)
+
+// NewDefaultConfig creates a new default configuration.
+//
+// Warning: it doesn't define mandatory `Token` and `DataCenter`
+// fields that need to be explicitly provided.
+func NewDefaultConfig() component.Config {
+	return &Config{}
+}
+
+// Validate checks the configuration for its validity.
+func (cfg *Config) Validate() error {
+	if cfg.DataCenter == "" && cfg.EndpointURLOverride == "" {
+		return ErrMissingDataCenter
+	}
+
+	if cfg.Insecure && cfg.EndpointURLOverride == "" {
+		return ErrInsecureInProd
```

Comment:
is this covered by tests? (please disregard if we drop the insecure config after all)

Reply:
It is now: https://github.com/solarwinds/solarwinds-otel-collector/pull/4/commits/53e7753aaf87cd430d131c5363ba430d54c7ad46 (just a single, simple testcase).